src/RestServer/Controllers/v1/LedgerController.cs

```diff
@@ -0,0 +1,387 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Neo.Network.P2P.Payloads;
+using Neo.Plugins.RestServer.Exceptions;
+using Neo.Plugins.RestServer.Extensions;
+using Neo.Plugins.RestServer.Models.Blockchain;
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.Plugins.RestServer.Models.Ledger;
+using Neo.SmartContract.Native;
+using System.Net.Mime;
+
+namespace Neo.Plugins.RestServer.Controllers.v1
+{
+    [Route("/api/v{version:apiVersion}/ledger")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class LedgerController : ControllerBase
+    {
+        private readonly NeoSystem _neosystem;
+
+        public LedgerController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+        }
+
+        #region Accounts
+
+        /// <summary>
+        /// Gets all the accounts that hold gas on the blockchain.
+        /// </summary>
+        /// <returns>An array of account details object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("gas/accounts", Name = "GetGasAccounts")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(AccountDetails[]))]
+        public IActionResult ShowGasAccounts()
+        {
+            var accounts = NativeContract.GAS.ListAccounts(_neosystem.StoreView, _neosystem.Settings);
```

Comment:
Consider mainnet and some public server providing this API. Won't it be easy to DOS it by doing these `accounts` requests?

Reply:
It would be something to test. I am using the same logic as `core`; that uses the `storage` `API`. But i do see your point.

Reply:
We can't have these methods unrestricted.